migrate/20250327_gen_ubid_uuid.rb

```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+
+Sequel.migration do
+  up do
+    # This function generates a pure-random (not timestamp-based) UUID,
+    # that is in valid UBID format.  It is passed the UBID type as an integer.
+    # You can get the type integer using `UBID.to_base32_n(prefix)`
+    # (e.g. `UBID.to_base32_n("vm") # => 884`).
+    #
+    # This could be used in the future as the DEFAULT value for uuid primary
+    # keys.
+    run <<~SQL
+      CREATE FUNCTION gen_random_ubid_uuid(ubid_type int) RETURNS uuid AS $$
+      DECLARE
+        r0 bigint;
+        r1 int;
+        r2 int;
+        r3 bigint;
+
+        p1 text;
+        p2 text;
+      BEGIN
+        -- 48 bit random
+        r0 = floor(281474976710656 * random()::numeric)::bigint;
+        -- 4 bit version + 2 bit random + 10 bit type
+        r1 = ((32 + floor(2 * random()::numeric)::integer) << 10) | ubid_type;
```

Comment:
Here we want 2 bits of random, but doesn't `floor(2 * random()::numeric)::integer` generate 0 or 1 (i.e. 1 bit of random)?